src/components/common/ColonyHome/ColonySubscription/ColonySubscription.tsx

```diff
@@ -81,35 +119,19 @@ const ColonySubscription = () => {
                 data-test="colonyMenuPopover"
                 title={MSG.colonyMenuTitle}
               />
-            )} */}
-            <div>***</div>
+            )}
           </ColonySubscriptionInfoPopover>
         )}
-        {!isSubscribed && (
+        {!canInteractWithColony && !walletConnecting && !initConnect && (
```

Comment:
Added `initConnect` to stop jumping in of Join button

Reply:
Perhaps we just remove the spinner above? We wouldn't need the additional `initConnect` logic in that case, and because we're just querying our own db, it's pretty fast and makes a spinner redunant. 

Testing with / without spinner and `initConnect` seem very similar, even when throttling to slow 3G.

Reply:
That said, leaving the spinner in and just commenting out `initConnect`  in `ColonySubscription`, I don't notice the Join button jumping. So not sure of the intention of `initConnect`.

Reply:
If you reload the app with auto login the `Join this Colony` flashes in and out, or if you connect a wallet. Basically, the async requests happening between `updateUser` and `updateWallet` means that there is a state where loading isn't happening but the user and wallet don't yet exist.

The `initConnect` was intended to bridge the the whole process and avoid this issue.

Reply:
@arrenv @willm30 
As far as I understand, what we want to achieve here is not show the Join button if wallet or user is loading (and if they're not already a watcher, but I'll leave it out).

We should be able to check for `walletConnecting` and `userLoading` and I tried that but was still experiencing the Join button flicker.

I think there's a bug in `AppContext`'s `updateUser`. We set `userLoading` to true, make the query, then set `userLoading` back to false, but we are not awaiting the query, hence userLoading is false even while the query is still loading.

Here's how I fixed this:
- Make `updateUser` async and await the `query` method
- Change `!initConnect` in the line above to `!userLoading` (destructure that from `useAppContext`)
- Remove `initConnect` from AppContext

Reply:
Thanks for looking into it @jakubcolony. I also feel we could do without `initConnect`. On master, `updateUser` is async as I added it some prs ago, so perhaps a rebase plus your suggestions here would solve the problem.